Packs/IBMStorageScale/Integrations/IBMStorageScale/IBMStorageScale.py

```diff
@@ -0,0 +1,546 @@
+
+        # Check if this event hash already exists
+        if event_hash in stored_hashes:
+            duplicates_found += 1
+            demisto.debug(f"Duplicate event found with hash {event_hash[:12]}...")
+            continue
+
+        # Add to deduplicated events and track new hash
+        deduplicated_events.append(event)
+        new_hashes[event_hash] = event_time
+
+    # Merge new hashes with stored hashes and store
+    all_hashes = {**stored_hashes, **new_hashes}
+    store_event_hashes(all_hashes)
+
+    stats = {
+        "total_events": len(events),
+        "duplicates_found": duplicates_found,
+        "unique_events": len(deduplicated_events),
+        "stored_hashes_count": len(all_hashes),
+    }
+
+    demisto.debug(f"Deduplication complete: {stats}")
+    return deduplicated_events, stats
+
+
+def get_fetch_start_time() -> datetime:
+    """
+    Get the start time for the fetch window from the last run object.
+    Returns a datetime object.
+    """
+    last_run = demisto.getLastRun()
+    last_fetch_time_str = last_run.get("last_fetch_time")
+
+    if last_fetch_time_str:
+        demisto.debug(f"Using last fetch time from last run: {last_fetch_time_str}")
+        # Parse ISO 8601 format, handling 'Z' for UTC
+        return datetime.fromisoformat(last_fetch_time_str.replace("Z", "+00:00"))
+    else:
+        # First run - use default lookback period based on DEFAULT_FIRST_FETCH_MINUTES (in days)
+        start_time = datetime.utcnow() - timedelta(days=DEFAULT_FIRST_FETCH_MINUTES)
+        demisto.debug(f"First run - using default lookback time: {start_time.isoformat()}Z")
+        return start_time
+
+
+def update_last_run_time(new_fetch_time: datetime) -> None:
+    """
+    Update the last run object with the given timestamp.
+    """
+    last_run = demisto.getLastRun()
+    # Store in UTC ISO format with 'Z', handling timezone-aware datetime
+    time_str = new_fetch_time.isoformat()
+    if not time_str.endswith("Z"):
+        time_str += "Z"
+    last_run["last_fetch_time"] = time_str
+    demisto.setLastRun(last_run)
+    demisto.debug(f"Updated last run with fetch time: {last_run['last_fetch_time']}")
+
+
+def generate_time_filter_regex(start_time: datetime, end_time: datetime) -> str:
+    """
+    Generates a regex for the 'entryTime' field to cover the given time window.
+    The regex will cover full minutes, relying on deduplication to handle overlaps.
+    """
+    # Floor the start time to the beginning of the minute
+    current_minute = start_time.replace(second=0, microsecond=0)
+
+    regex_parts = []
+
+    # Iterate minute by minute through the time window
+    while current_minute <= end_time:
+        # Format the minute prefix, e.g., "2025-08-07T14:30"
+        minute_prefix = current_minute.strftime("%Y-%m-%dT%H:%M")
+        # Create a regex for all 60 seconds within that minute
+        regex_parts.append(f"{minute_prefix}:[0-5][0-9]")
```

Comment:
consider extracting date format and regex format to a function or const with a comment about why this was chosen + UT